lib/exceptions/CustomerWithThatEmailAlreadyExists.ts

```diff
@@ -0,0 +1,9 @@
+import HttpException from "./HttpException";
+
+class UserWithThatEmailAlreadyExistsException extends HttpException {
```

Comment:
Wofür baust du hier extra eine Klasse auf? Sieht für mich ein bisschen nach overengineering aus ?

Reply:
Wäre es nicht möglich eine Klasse mit den ganzen exceptions zu erstellen. Statt für jede exception eine eigene Klasse. Wie du das machst erinnert mich das ein bisschen an Java Enterprise lösung ? :D

Reply:
Evt. in Richtung einer factory oder eines decorator patterns?

Reply:
muss ich mal drüber nachdenken, aber iwie muss ich im aufrufenden Code halt explizit sagen, dass der User keine Mail hat. Wüsste net ganz wie ich das noch kürzen könnte
 Ist halt weird, dass es einen eigenen File bekommt, zur Not könnte man auch verwandte Exceptions in ein File schreiben.

Reply:
Warum musst du den für jede Exception eine eigene Klasse aufmachen? Die erben doch alle von HttpException oder nicht?

Reply:
naja, was ist der unterschied dann? Bei Klassen hab ich die Autocomplete dabei, wenn ich den konkreten Grund reinschreibe, habe ich im Zweifel typos drin und es wird uneinheitlich.

Reply:
okay 👍

Reply:
je nachdem wie viele Fehlerfälle du hast, wirds halt dann nur ziemlich aufgebläht und unübersichtlich.